fix: audit fixes

gabririgo opened the pull request:
- use `swapDelta` instead of `amountSpecified` for backrun amount
- correctly retrieve original sender for ERC6909 credit from `msgSender()` method in swap router
- update tests

gabririgo committed:
fix: use swap delta instead of amount specified for backrun

- use swap delta amount0 as basis for backrun amount, as swap could be partially executed in case of a big swap with a small liquidity pool

gabririgo committed:
fix: read msgSender() from router

- retrieve original sender from router and credit backrun amount to this address
- revert if used router does not implement msgSender()
- update tests

leeftk commented on src/BackGeoOracle.sol:
```diff
@@ -206,11 +208,11 @@ contract BackGeoOracle is BaseHook {
             int128 numerator = int128(tickDelta) * 9999;
             int128 denominator = int128(Oracle.LIMIT_ABS_TICK_MOVE) * 10000;
-            params.amountSpecified = params.amountSpecified * numerator / denominator;
+            params.amountSpecified = swapDelta.amount0() * numerator / denominator;
```
LGTM!

leeftk commented on src/BackGeoOracle.sol:
```diff
@@ -161,8 +169,17 @@ contract BackGeoOracle is BaseHook {
+            address backrunAmountRecipient;
+
+            // retrieve original sender. The calling router is expected to implement msgSender() method
+            try IMsgSender(sender).msgSender() returns (address originalSender) {
```
Still reviewing this with Vinicius to see what his thoughts are. Mine personally are that I like this solution, it's elegant, puts some of the responsibility on the user only to use a verified router, and removes that centralization vector of adding an owner. He also had some thoughts, but I asked him to add them directly, so he'll add them here when he can. As for me, this is a good fix.

leeftk reviewed (commented):
Added some comment but overall LGTM and is good to merge.

viniciuscoassis commented:
Yes, I like the solution, I think it will work. Theres still the thing with routers and native eth, but if we don't want to add ownership to the hook nor add complexity to integrators, this will indeed be the best solution.

leeftk commented on src/BackGeoOracle.sol:
```diff
@@ -15,6 +15,10 @@ import {PoolId, PoolIdLibrary} from "v4-core/src/types/PoolId.sol";
+interface IMsgSender {
+    function msgSender() external view returns (address);
+}
+
```
Small suggestion but adding to the natspec here that only trusted and verified rotuers should be used could be helpful to users and developers as well interacting with this pool/hook.

gabririgo committed:
docs: add natspec docs for contract

- define contract and some aspects critical to integrators (i.e. swap routers must implement `msgSender()` and ERC6909 recipient is expected to be able to burn ERC6909 balances)

leeftk commented:
LGTM!
